operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -0,0 +1,401 @@
+use std::collections::HashSet;
+use std::marker::PhantomData;
+
+use futures::stream;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+
+use geoengine_datatypes::collections::{
+    BuilderProvider, DataCollection, FeatureCollection, FeatureCollectionInfos,
+    FeatureCollectionModifications, FeatureCollectionRowBuilder, GeoFeatureCollectionRowBuilder,
+    IntoGeometryIterator,
+};
+use geoengine_datatypes::primitives::{DataRef, FeatureDataRef, Geometry};
+use geoengine_datatypes::util::arrow::ArrowTyped;
+
+use crate::adapters::FeatureCollectionChunkMerger;
+use crate::engine::{QueryContext, QueryProcessor, QueryRectangle, VectorQueryProcessor};
+use crate::error::Error;
+use crate::util::Result;
+
+/// Implements a left inner join between a `GeoFeatureCollection` stream and a `DataCollection` stream.
+pub struct EquiLeftJoinProcessor<G> {
+    vector_type: PhantomData<FeatureCollection<G>>,
+    left_processor: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+    right_processor: Box<dyn VectorQueryProcessor<VectorType = DataCollection>>,
+    left_column: String,
+    right_column: String,
+    right_column_prefix: String,
+}
+
+impl<G> EquiLeftJoinProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+    for<'g> FeatureCollection<G>: IntoGeometryIterator<'g>,
+    for<'g> <FeatureCollection<G> as IntoGeometryIterator<'g>>::GeometryType: Into<G>,
+    FeatureCollectionRowBuilder<G>: GeoFeatureCollectionRowBuilder<G>,
+{
+    pub fn new(
+        left_processor: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+        right_processor: Box<dyn VectorQueryProcessor<VectorType = DataCollection>>,
+        left_column: String,
+        right_column: String,
+        right_prefix: String,
+    ) -> Self {
+        Self {
+            vector_type: PhantomData,
+            left_processor,
+            right_processor,
+            left_column,
+            right_column,
+            right_column_prefix: right_prefix,
+        }
+    }
+
+    fn join<'c>(
+        &self,
+        left: &'c FeatureCollection<G>,
+        right: &DataCollection,
```

Comment:
why do left and right collection have different lifetimes?

Reply:
They are produced by different streams. I just made it as relaxed as possible.

Reply:
It is now quite different
7d53833